### PR TITLE
[Doc] Update ST-Link doc following debugging chapter introduction

### PR DESCRIPTION
**Pull request by erwango**
Following #14000, a debugging section is introduced to describe properly debug probes and matching host tools. In this context it makes sense to remove existing doc/guides/tools/stlink-v21.rst and dispatch information in newly created files.
Remains some specific information, dedicated to OpenOcd/JLink usage with STLink which hardly fits in host-tool.rst or probes.rst, so create a new file stlink-v21.rst in debugging section. I'm open to other option to have that allow to have this information centralized in another place.

Based on #14000.

### CODE REVIEW

**Review by dbkinder — CHANGES_REQUESTED** [review]
Looks OK, thanks.

**Comment by erwango** [comment]
@dbkinder 
> Looks OK, thanks.

Well, can you approve then ? ;-)

**Review by dbkinder — APPROVED** [review]
Thought I had approved. Try again.

File: doc/guides/debugging/host-tools.rst [hunk 1]
```diff
@@ -54,6 +54,7 @@ These debug host tools are compatible with the following debug probes:
 
 - :ref:`opensda-daplink-onboard-debug-probe`
 - :ref:`jlink-external-debug-probe`
+- :ref:`stlink-v21-onboard-debug-probe`
```
Comment (MaureenHelm) [inline doc/guides/debugging/host-tools.rst]: Do you need to add a similar line to the J-Link host tools section?

File: doc/guides/debugging/probes.rst [hunk 2]
```diff
@@ -137,6 +137,31 @@ As with all OpenSDA debug probes, the steps for programming the firmware are:
...
+
+ST-LINK/V2-1 is a serial and debug adapter built into all Nucleo and Discovery
+boards. It provides a bridge between your computer (or other USB host) and the
+embedded target processor, which can be used for debugging, flash programming,
+and serial communication, all over a simple USB cable.
+
+It is compatible with the following host debug tools:
+
+- :ref:`openocd-debug-host-tools`
+- :ref:`jlink-debug-host-tools`
```
Comment (MaureenHelm) [inline doc/guides/debugging/probes.rst]: I think pyOCD recently added support for ST-Link

File: doc/guides/debugging/host-tools.rst [hunk 3]
```diff
@@ -88,3 +97,6 @@ Check if your SoC is listed in `pyOCD Supported Devices`_.
 
 .. _pyOCD Supported Devices:
    https://github.com/mbedmicro/pyOCD/tree/master/pyocd/target/builtin
+
+.. _OpenOCD Windows:
+    http://gnutoolchains.com/arm-eabi/openocd/
```
Comment (MaureenHelm) [inline doc/guides/debugging/host-tools.rst]: Why here? What about the GNU MCU Eclipse mentioned in the "Getting OpenOCD" page?

File: doc/guides/debugging/index.rst [hunk 4]
```diff
@@ -8,3 +8,4 @@ Debugging
 
    host-tools.rst
    probes.rst
+   stlink-v21.rst
```
Comment (MaureenHelm) [inline doc/guides/debugging/index.rst]: Is this separate page really still necessary?

File: doc/guides/debugging/probes.rst (hunk 2 above)
Comment (erwango) [reply doc/guides/debugging/probes.rst]: Indeed, but not widely deployed so far. This is worth adding a comment anyway.

File: doc/guides/debugging/host-tools.rst (hunk 1 above)
Comment (erwango) [reply doc/guides/debugging/host-tools.rst]: Yeah, will do

File: doc/guides/debugging/host-tools.rst (hunk 3 above)
Comment (erwango) [reply doc/guides/debugging/host-tools.rst]: Seems this may not be the best way to get up to date openocd binaries (for windows at least). For windows, it redirects to http://www.freddiechopin.info/en/download/category/4-openocd with latest build having: 'build date: 2017-01-23'. (which is older than the oldest binary in http://gnutoolchains.com/arm-eabi/openocd/)

File: doc/guides/debugging/index.rst (hunk 4 above)
Comment (erwango) [reply doc/guides/debugging/index.rst]: Information provided in this page is not available anywhere else so it's worth keeping, IMHO. This being said I'm open to alternative locations.

Comment (MaureenHelm) [reply doc/guides/debugging/index.rst]: I think it can be worked into the debug probe page

Comment (erwango) [reply doc/guides/debugging/index.rst]: Ok, done!

File: doc/guides/debugging/probes.rst [hunk 5]
```diff
@@ -137,6 +137,97 @@ As with all OpenSDA debug probes, the steps for programming the firmware are:
...
+firmware please visit: `Segger over ST-Link`_
+
+Flash and debug with ST-Link
+============================
+
+Using Openocd
+-------------
+
+Openocd is available by default on ST-Link and configure as default flash and
+debug tool, flash and debug could be done as follows:
```
Comment (MaureenHelm) [inline doc/guides/debugging/probes.rst]: Suggest rewording a little:
```
OpenOCD is available by default on ST-Link and configured as the default flash and
debug tool. Flash and debug can be done as follows:
```

File: doc/guides/debugging/probes.rst [hunk 6]
```diff
@@ -137,6 +137,97 @@ As with all OpenSDA debug probes, the steps for programming the firmware are:
...
+most J-Link features like the ultra fast flash download and debugging speed or
+the free-to-use GDBServer.
+
+More informations about upgrading ST-LINK/V2-1 to JLink or restore ST-Link/V2-1
+firmware please visit: `Segger over ST-Link`_
+
+Flash and debug with ST-Link
+============================
+
+Using Openocd
```
Comment (MaureenHelm) [inline doc/guides/debugging/probes.rst]: OpenOCD

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: doc/guides/debugging/probes.rst (hunk 5 above)
Comment (erwango) [reply doc/guides/debugging/probes.rst]: Txs, fixed

File: doc/guides/debugging/probes.rst (hunk 6 above)
Comment (erwango) [reply doc/guides/debugging/probes.rst]: Fixed

**Review by MaureenHelm — APPROVED** [review]
